### PR TITLE
Added a decorator function to validate parameters 

### PR DESCRIPTION
**Pull request by shantanuo**
read_excel method needs to check if it has received valid parameters only. Use of **kwds allows any parameter that leads to confusion and bugs.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Take a look at pandas.util._validators.validate_kwargs

**Comment by WillAyd** [comment]
First and foremost this needs tests.

If this is only for `read_excel` then I think the decorator is overkill - would be better served directly within the function body. 

Is this in reference a particular issue #? If so can you update your post to link to it?

**Comment by shantanuo** [comment]
This is not only related to "read_excel" function. All read_* methods accept *kwargs due to which non-valid parameters are allowed. 

https://github.com/pandas-dev/pandas/issues/22189

This is the only issue that is holding me back from using pandas and therefore I tried to fix it :)

File: pandas/util/_decorators.py [hunk 1]
```diff
@@ -5,6 +5,21 @@
 from textwrap import dedent, wrap
 from functools import wraps, update_wrapper, WRAPPER_ASSIGNMENTS
 
+def validate_kwarg():
+    def _validate_kwarg(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            expected_keys=['io', 'sheet_name','header', 'names', 'index_col', 'usecols', 'squeeze', 'dtype', 'engine',
```
Comment (WillAyd) [inline pandas/util/_decorators.py]: Just instantiate this as a set

File: pandas/util/_decorators.py [hunk 2]
```diff
@@ -5,6 +5,21 @@
...
+def validate_kwarg():
+    def _validate_kwarg(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            expected_keys=['io', 'sheet_name','header', 'names', 'index_col', 'usecols', 'squeeze', 'dtype', 'engine', 
+               'converters', 'true_values',  'false_values', 'skiprows', 'nrows', 'na_values', 'verbose', 'parse_dates',
+               'date_parser', 'thousands', 'comment', 'skipfooter', 'convert_float']
+
+            if set(kwargs.keys()).difference(set(expected_keys)): 
+                raise ValueError('invalid parameter found')
```
Comment (WillAyd) [inline pandas/util/_decorators.py]: Would be nice to be explicit about which parameter isn't working. Also want a test

**Review by WillAyd — CHANGES_REQUESTED** [review]
I still think the wrapper is overcomplicating things. As mentioned before tests are the most critical thing here and ideally should come before any other coding changes

**Comment by TomAugspurger** [comment]
I think we should focus our efforts on removing the need for `**kwargs` from the signature, rather than trying to work around it.

**Comment by shantanuo** [comment]
sure. Decorator was my shortcut to incorporate the functionality that I needed desperately :)

**Comment by jbrockmendel** [comment]
Am I missing something?  Why is the existing validate_kwargs decorator not up to the task?

**Comment by shantanuo** [comment]
I checked the existing validate_kwargs decorator, but could not figure out how to use it :) 
Removing **kwargs seems to be ideal though. Any one of the two (that is practical, easy and quick) approach should be adopted.

**Comment by jbrockmendel** [comment]
@shantanuo any plans to add a test to this?  I doubt this implementation will be merged, but a test could help us figure out how to use the existing decorators for this purpose.

**Comment by jreback** [comment]
closing as stale. if you'd like to continue, pls ping.

**Pull request closed without merging**